code/modules/projectiles/guns/projectile/CMturret.dm

```diff
@@ -0,0 +1,228 @@
+/mob/living/carbon/human
+	var/obj/structure/machinegun/mounted
+
+	ClickOn(var/atom/A, params)
+		if(mounted)
+			if(mounted.loc == src.loc)
+				if(A && mounted.nextshot <= world.time && mounted.anchored)
+					mounted.shoot(get_turf(A))
+			else
+				mounted = null
+		else
+			..()
+
+
+/obj/structure/machinegun
+	name = "machine gun"
+	desc = "A stationary machine gun."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "mgun+barrier"
+	var/fire_sound = 'sound/weapons/Gunshot_smg.ogg'
+	var/empty_sound = 'sound/weapons/empty.ogg'
+	var/ammo_type = "/obj/item/projectile/bullet/a762"
+	var/ammo = 500
+	var/ammomax = 500
+	var/list/row1 = list()
+	var/list/row2 = list()
+	var/list/row3 = list()
+	var/mob/living/carbon/human/user
+	var/nextshot = 0
+	var/FIRETIME = 1 //tenths of seconds
+	density = 1
+	anchored = 0
+	flags = ON_BORDER
+	var/health = 500
+
+/obj/structure/machinegun/process()
+	var/mob/living/carbon/human/U
+	for(var/mob/living/carbon/human/H in range(0, src))
```

Comment:
I just ported it, honestly.
That said, I'm not going to really do this until tomorrow, so if you want it adjusted, tell me what to put down and I'll add it to the list of things that I need to fix with it.